File: OCP/Spoilers/MarsRoverKata/Commands/FireMortarCommand.cs
namespace MarsRoverKata.Commands
{
    public class FireMortarCommand : ICommand
    {
        private Rover Rover { get; set; }

        public FireMortarCommand(Rover rover)
        {
            Rover = rover;
        }

        public bool Execute()
        {
            return Rover.FireProjectile(typeof(Mortar));
        }
    }
}

File: LSP+ISP/Java/MarsRoverKata/src/main/java/MarsRoverKata/Commands/FireMortarCommand.java
package MarsRoverKata.Commands;

import MarsRoverKata.Mortar;
import MarsRoverKata.Rover;

public class FireMortarCommand implements ICommand {
    private Rover rover;

    public Rover getRover() {
        return rover;
    }

    public void setRover(Rover rover) {
        this.rover = rover;
    }

    public FireMortarCommand(Rover rover) {
        this.rover = rover;
    }

    public boolean execute() {
        return rover.fireProjectile(Mortar.class);
    }
}

